analysis_options.yaml

```diff
@@ -0,0 +1,63 @@
+analyzer:
+  strong-mode: true
+    # implicit-dynamic: false
+    # implicit-casts: false
+  exclude:
+    - bazel-*
+
+linter:
+  rules:
+    - always_declare_return_types
+    # - always_specify_types - needed for having shorthand final / var
+    - annotate_overrides
+    # - avoid_as - needed for strong mode
+    - avoid_empty_else
+    - avoid_init_to_null
+    - avoid_return_types_on_setters
+    - await_only_futures
+    - camel_case_types
+    - cancel_subscriptions
+    - close_sinks
+    # - constant_identifier_names - very minor.
+    - control_flow_in_finally
+    - directives_ordering
+    - empty_catches
+    - empty_constructor_bodies
+    - empty_statements
+    - hash_and_equals
+    - implementation_imports
+    - invariant_booleans
+    - iterable_contains_unrelated_type
+    - library_names
+    - library_prefixes
+    - literal_only_boolean_expressions
+    - list_remove_unrelated_type
+    - no_duplicate_case_values
+    - non_constant_identifier_names
+    # - one_member_abstracts - better object oriented code.
```

Comment:
[nit] is this really better? In my experience things do get more readable with a typdef...

Reply:
Honestly I just copied these over from the old analysis_options.yaml and haven't really modified these. I'm uncommenting this one for now.